### PR TITLE
add support for RFC5914 structs. 

### PR DESCRIPTION
**Pull request by carl-wallace**
make adjusts to some prior structure definitions to work around macro-related issues. Notes on non-use of macros

In trust_anchor_format.rs
- Choice was not used on TrustAnchorChoice owing to types not available in Asn1Type enum
- Sequence was not used on TrustAnchorInfo owing to lack of DecodeValue required by context_specific

error[E0277]: the trait bound `TrustAnchorInfo<'a>: DecodeValue<'_>` is not satisfied
   --> x509/src/trust_anchor_format.rs:295:22
    |
295 |                     .context_specific::<TrustAnchorInfo<'a>>(TAC_TA_INFO_TAG, TagMode::Explicit)?;
    |                      ^^^^^^^^^^^^^^^^ the trait `DecodeValue<'_>` is not implemented for `TrustAnchorInfo<'a>`
    |
note: required by a bound in `Decoder::<'a>::context_specific`
   --> der/src/decoder.rs:172:12
    |
172 |         T: DecodeValue<'a> + FixedTag,
    |            ^^^^^^^^^^^^^^^ required by this bound in `Decoder::<'a>::context_specific`

- Sequence was not used on CertPathControls to use decode_implicit for each field (trailing bits were left using the default decoder even with mods to Certificate and NameConstraints)

In certificate.rs
- Sequence was not used on Certificate due to same issue as described above for TrustAnchorInfo (i.e., tension between use of DecodeValue and Decodable)
- Sequence was not used on NameConstraints for the same DecodeValue vs Decodable issue.

### CODE REVIEW

File: x509/src/certificate.rs [hunk 1]
```diff
@@ -202,6 +202,33 @@ pub struct Certificate<'a> {
...
+    where
+        F: FnOnce(&[&dyn der::Encodable]) -> ::der::Result<T>,
+    {
+        f(&[
+            &self.tbs_certificate,
+            &self.signature_algorithm,
+            &self.signature,
+        ])
+    }
+}
```
Comment (tarcieri) [inline x509/src/certificate.rs:230]: Why doesn't custom derive work here?

Comment (carl-wallace) [reply x509/src/certificate.rs:230]: The tagged field in CertPathControls generated that need. Same thing for NameConstraints (sibling tagged field in CertPathControls). Likely the same issue underpins the OCSP issue I mentioned earlier today as well (where GeneralName is the target).

Comment (carl-wallace) [reply x509/src/certificate.rs:230]: If you put Certificate, NameConstraints and CertPathControls back with Sequence derive and corresponding context-specific attributes (which is how they started), the following error results:

```
error[E0277]: the trait bound `certificate::Certificate<'_>: DecodeValue<'_>` is not satisfied
   --> x509/src/trust_anchor_format.rs:146:39
    |
146 | #[derive(Clone, Debug, Eq, PartialEq, Sequence)]
    |                                       ^^^^^^^^ the trait `DecodeValue<'_>` is not implemented for `certificate::Certificate<'_>`
    |
note: required by a bound in `der::asn1::ContextSpecific::<T>::decode_implicit`
   --> der/src/asn1/context_specific.rs:71:12
    |
71  |         T: DecodeValue<'a> + Tagged,
    |            ^^^^^^^^^^^^^^^ required by this bound in `der::asn1::ContextSpecific::<T>::decode_implicit`
    = note: this error originates in the derive macro `Sequence` (in Nightly builds, run with -Z macro-backtrace for more info)
```
```
error[E0277]: the trait bound `pkix_extensions::NameConstraints<'_>: DecodeValue<'_>` is not satisfied
   --> x509/src/trust_anchor_format.rs:146:39
    |
146 | #[derive(Clone, Debug, Eq, PartialEq, Sequence)]
    |                                       ^^^^^^^^ the trait `DecodeValue<'_>` is not implemented for `pkix_extensions::NameConstraints<'_>`
    |
note: required by a bound in `der::asn1::ContextSpecific::<T>::decode_implicit`
   --> der/src/asn1/context_specific.rs:71:12
    |
71  |         T: DecodeValue<'a> + Tagged,
    |            ^^^^^^^^^^^^^^^ required by this bound in `der::asn1::ContextSpecific::<T>::decode_implicit`
    = note: this error originates in the derive macro `Sequence` (in Nightly builds, run with -Z macro-backtrace for more info)

```

Comment (carl-wallace) [reply x509/src/certificate.rs]: It worked and changes were pushed to catch up #356.

**Comment by tarcieri** [comment]
@carl-wallace can you rebase? I just want to make sure #377 didn't cause any breakages

**Comment by carl-wallace** [comment]
I pushed changes to use x501 in the trust_anchor_format file.

File: x509/src/trust_anchor_format.rs [hunk 2]
```diff
@@ -0,0 +1,268 @@
...
+            version,
+            pub_key,
+            key_id,
+            ta_title,
+            cert_path,
+            extensions,
+            ta_title_lang_tag,
+        })
+    }
+}
```
Comment (tarcieri) [inline x509/src/trust_anchor_format.rs:77]: Can this one use the proc macro now?

**Comment by tarcieri** [comment]
In the interest of moving things along I'm going to go ahead and merge this.

Perhaps @npmccallum can take a look at converting them to a proc macro in #363.

File: x509/src/trust_anchor_format.rs (hunk 2 above)
Comment (tarcieri) [reply x509/src/trust_anchor_format.rs:77]: I guess you left a note here:

> `Sequence` was not used on `TrustAnchorInfo` owing to lack of `DecodeValue` required by `context_specific`

It should probably work now.

**Pull request merged**